src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/repository/TaskResourceRepositoryTest.java

```diff
@@ -233,17 +391,167 @@ void given_task_is_created_when_count_by_role_signature_then_correct_count_retur
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId2, taskIds.get(0));
+    }
+
+    @Test
+    void given_tasks_created_when_search_request_received_then_task_ids_are_returned_and_filter_by_role_signature() {
+        String taskId2 = UUID.randomUUID().toString();
+        TaskResource createdTask = createTask(taskId2, "case-manager", "WA",
+            "reviewAppeal", "anotherAssignee", "1623278362430413", CFTTaskState.ASSIGNED);
+
+        transactionHelper.doInNewTransaction(() -> {
+            task.setIndexed(true);
+            createdTask.setIndexed(true);
+            taskResourceRepository.save(task);
+            taskResourceRepository.save(createdTask);
+        });
+
+        Set<String> filterSignature = Set.of("*:WA:*:*:1:765324", "*:IA:*:*:1:765324");
+        Set<String> roleSignature = Set.of("WA:*:*:case-manager:*:r:U:*");
+        SearchRequest request = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED))
+            .caseIds(List.of("1623278362430412", "1623278362430413"))
+            .taskTypes(List.of("startAppeal", "reviewAppeal"))
+            .users(List.of("someAssignee", "anotherAssignee"))
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId2, taskIds.get(0));
+    }
+
+    @Test
+    void given_tasks_created_when_search_request_for_available_task_then_task_ids_without_assignee_are_returned() {
+        String taskId2 = UUID.randomUUID().toString();
+        TaskResource createdTask = createTask(taskId2, "case-manager", "IA",
+            "reviewAppeal", null, "1623278362430413", CFTTaskState.UNASSIGNED);
+
+        transactionHelper.doInNewTransaction(() -> {
+            task.setIndexed(true);
+            createdTask.setIndexed(true);
+            taskResourceRepository.save(task);
+            taskResourceRepository.save(createdTask);
+        });
+
+        Set<String> filterSignature = Set.of("*:IA:*:*:1:765324");
+        Set<String> roleSignature = Set.of("IA:*:*:tribunal-caseofficer:*:r:U:*", "IA:*:*:case-manager:*:r:U:*");
+        SearchRequest request = SearchRequest.builder()
+            .requestContext(RequestContext.AVAILABLE_TASKS)
+            .caseIds(List.of("1623278362430412", "1623278362430413"))
+            .taskTypes(List.of("startAppeal", "reviewAppeal"))
+            .users(List.of("someAssignee", "anotherAssignee"))
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId2, taskIds.get(0));
```

Comment:
assert assignee is null?

Reply:
Done in RWA-2212 in CFTTaskDatabaseServiceSearchTest